File: microservers/billetterieservice/src/main/java/io/multi/billetterieservice/domain/Billet.java
package io.multi.billetterieservice.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Billet {

    private Long billetId;
    private String billetUuid;
    private Long commandeId;
    private String codeBillet;
    private String numeroSiege;
    private String nomPassager;
    private String telephonePassager;
    private String pieceIdentite;
    private String statut;
    private OffsetDateTime dateValidation;
    private Long validePar;
    private String qrCodeData;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}

File: mobile_app/lib/models/billet.dart
class Billet {
  final String? billetUuid;
  final String codeBillet;
  final String? numeroSiege;
  final String nomPassager;
  final String? telephonePassager;
  final String? pieceIdentite;
  final String statut;
  final String? qrCodeData;
  final String? createdAt;

  Billet({
    this.billetUuid,
    required this.codeBillet,
    this.numeroSiege,
    required this.nomPassager,
    this.telephonePassager,
    this.pieceIdentite,
    this.statut = 'ACTIF',
    this.qrCodeData,
    this.createdAt,
  });

  /// Data to encode in QR code: qrCodeData if set by backend, otherwise codeBillet.
  String get qrContent =>
      qrCodeData?.isNotEmpty == true ? qrCodeData! : codeBillet;

  factory Billet.fromJson(Map<String, dynamic> json) {
    return Billet(
      billetUuid: json['billetUuid'],
      codeBillet: json['codeBillet'] ?? '',
      numeroSiege: json['numeroSiege'],
      nomPassager: json['nomPassager'] ?? '',
      telephonePassager: json['telephonePassager'],
      pieceIdentite: json['pieceIdentite'],
      statut: json['statut'] ?? 'ACTIF',
      qrCodeData: json['qrCodeData'],
      createdAt: json['createdAt']?.toString(),
    );
  }
}
